src/types/artifact.ts

```diff
@@ -16,6 +16,14 @@ export function isMarkdownArtifactData(data: ArtifactData): data is MarkdownArti
   return typeof data === 'string'
 }
 
-export function isTableArtifactData(data: ArtifactData): data is TableArtifactData {
-  return Array.isArray(data)
+export function isArrayOfMaps(data: unknown): data is Record<string, unknown>[] {
+  return Array.isArray(data) && data.every(row => typeof row === 'object')
```

Comment:
Do we need to exclude arrays and nulls here? In the every

Reply:
Ah yeah we do - good catch

Reply:
I think there's an isRecord utility for this

Reply:
nice, good looks 👍🏻